_posts/2020-01-03-graphql-custom-directives.markdown

```diff
@@ -0,0 +1,266 @@
+---
+layout: epic
+title: Better GraphQL Error Handling I - Using Custom Directives
+date: 2020-01-03
+categories: [GraphQL, Error Handling]
+author: matt
+---
+
+This will be the first in a series of posts about how we used some more advanced GraphQL tooling and functionality
+to better handle errors occurring during query resolution, and better equip clients to reason about such errors.
+
+The goal is to describe our current approach, but also do a deep dive into specific ways we've extended our
+[GraphQL server](artsy/metaphysics) to help us accomplish that. Therefore, if you are an interested GraphQL user,
+you may find this useful, even if some of the larger context specifically around how we are using it to help
+standardize error handling doesn't apply.
+
+<!-- more -->
+
+## Introduction and Initial Context
+
+At Artsy, we use GraphQL as our API language of choice. In particular, [as we've described before](link to prev
+blogposts), we have an orchestration layer speaking GraphQL, which is what our front-end clients talk to. The
+GraphQL orchestration layer wraps up access to several backend services, which are made accessible via a
+combination of data loaders and [GraphQL stitching](apollo link). [Apollo Federation](apoll link) is another tool
+people are using to bring together disaparte backends when using GraphQL in an orchestration layer.
+
+Now, consider the following query, which is a realistic one you might see when accessing a 'product' page.
+
+```
```

Comment:
these ticks can accept a graphql type to highlight text (including down below)

Reply:
Interesting! Is that the case? Looks like the list of [supported languages](https://pygments.org/languages/) doesn't include GraphQL, and other posts in the blog don't use tags for GraphQL fragments.

Will update other snippets here tho (typescript, json, etc)